test/extended/images/hardprune.go

```diff
@@ -475,22 +475,39 @@ func ConfigureRegistry(oc *exutil.CLI, desiredState RegistryConfiguration) error
 
 	err = oc.Run("env").Args(append([]string{"dc/docker-registry"}, envOverrides...)...).Execute()
 	if err != nil {
-		return fmt.Errorf("failed to update registry's environment with %s: %v", &waitForVersion, err)
+		return false, fmt.Errorf("failed to update registry's environment with %s: %v", &waitForVersion, err)
 	}
-	return exutil.WaitForDeploymentConfig(
+	err = exutil.WaitForDeploymentConfig(
 		oc.AdminKubeClient(),
 		oc.AdminAppsClient().Apps(),
 		metav1.NamespaceDefault,
 		"docker-registry",
 		waitForVersion,
 		oc)
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func RedeployRegistry(oc *exutil.CLI) (bool, error) {
+	defer func(ns string) { oc.SetNamespace(ns) }(oc.Namespace())
+
+	oc = oc.SetNamespace(metav1.NamespaceDefault).AsAdmin()
+
+	_, err := oc.Run("rollout").Args("latest", "dc/docker-registry", "--again=true").Output()
```

Comment:
not sure.  Will this cancel the existing rollout?   what's waiting for this rollout to complete before running tests?

Reply:
We are waiting for the completion of the rollout when reconfigure the server:

https://github.com/legionus/origin/blob/1e8bbac9b1184e471c9ffecbb31ba285d5b8c662/test/extended/images/hardprune.go#L480-L490

The `oc rollout latest` will not wait for the rollout to be completed?

Reply:
If not, I can add the same check.

Reply:
you can ask @mfojtik but i do not believe so, i think it just triggers the rollout, it doesn't wait for it to finish.

Reply:
OK. I will check it.

Reply:
you just need to be careful because i'm not sure what happens if you trigger a rollout when a rollout is in progress..  Suppose you have the following situation:

1) current deployment is v1
2) a deployment is running to create v2
3) you check the current deploy version and see v1
4) you invoke rollout (which will eventually create v3, but it's queued to happen after v2 finishes)
5) you wait for v1+1 (v2)
6) the running deployment finishes, you start testing
7) the v3 deployment kicks off and interrupts your test.

again probably need @mfojtik to confirm what happens at (4).

Reply:
oc rollout will not wait, you have to run "oc rollout status" to wait.